kyaml/filesys/filesystem_test.go

```diff
@@ -42,3 +52,87 @@ func testNotExistErr(t *testing.T, fs FileSystem) {
 	err = fs.Walk(path, func(_ string, _ os.FileInfo, err error) error { return err })
 	assert.Truef(t, errors.Is(err, os.ErrNotExist), "Walk should return ErrNotExist, got %v", err)
 }
+
+// setupFileSys returns file system, default directory to test in, and working directory
+func setupFileSys(t *testing.T, name string) (FileSystem, string, string) {
+	t.Helper()
+
+	switch name {
+	case disk:
+		fSys, testDir := makeTestDir(t)
+		return fSys, testDir, cleanWd(t)
+	case memoryAbs:
+		return filesysBuilders[name](), Separator, Separator
+	default:
+		require.Equal(t, memoryEmpty, name)
+		return filesysBuilders[name](), "", ""
+	}
+}
+
+func TestConfirmDir(t *testing.T) {
+	for name := range filesysBuilders {
+		name := name
+		t.Run(name, func(t *testing.T) {
+			fSys, prefixPath, wd := setupFileSys(t, name)
+
+			d1Path := filepath.Join(prefixPath, "d1")
+			d2Path := filepath.Join(d1Path, ".d2")
+			err := fSys.MkdirAll(d2Path)
+			require.NoError(t, err)
+			require.Truef(t, fSys.Exists(d2Path), existMsg, d2Path)
+
+			tests := map[string]*struct {
+				dir      string
+				cleanDir string
```

Comment:
```suggestion
				expected string
```

It is easier for any readers of this code to understand tests when you name the expected value `expected`